x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/perform_rule_upgrade/perform_rule_upgrade_route.schema.yaml

```diff
@@ -0,0 +1,865 @@
+            history_window_start:
+              oneOf:
+                - type: object
+                  required:
+                    - pick_version
+                  properties:
+                    pick_version:
+                      $ref: '#/components/schemas/PickVersionValues'
+                - type: object
+                  required:
+                    - pick_version
+                    - resolved_value
+                  properties:
+                    pick_version:
+                      type: string
+                      enum: [RESOLVED]
+                    resolved_value:
+                      $ref: '../../model/rule_schema/specific_attributes/new_terms_attributes.schema.yaml#/components/schemas/HistoryWindowStart'
+
+
+
+    UpgradeSpecificRulesRequest:
+      type: object
+      required:
+        - mode
+        - rules
+      properties:
+        mode:
+          type: string
+          enum: [SPECIFIC_RULES]
+        rules:
+          type: array
+          items:
+            $ref: '#/components/schemas/RuleUpgradeSpecifier'
+        pick_version:
+          $ref: '#/components/schemas/PickVersionValues'
+
+    UpgradeAllRulesRequest:
+      type: object
+      required:
+        - mode
+      properties:
+        mode:
+          type: string
+          enum: [ALL_RULES]
+        pick_version:
+          $ref: '#/components/schemas/PickVersionValues'
+
+    SkipRuleUpgradeReason:
+      type: string
+      enum: [RULE_UP_TO_DATE, RULE_NOT_FOUND]
+
+    SkippedRuleUpgrade:
+      type: object
+      required:
+        - rule_id
+        - reason
+      properties:
+        rule_id:
+          type: string
+        reason:
+          $ref: '#/components/schemas/SkipRuleUpgradeReason'
+
+    PerformRuleUpgradeResponseBody:
+      type: object
+      required:
+        - summary
+        - results
+        - errors
+      properties:
+        summary:
+          type: object
+          required:
+            - total
+            - succeeded
+            - skipped
+            - failed
+          properties:
+            total:
+              type: number
+            succeeded:
+              type: number
+            skipped:
+              type: number
+            failed:
+              type: number
+        results:
+          type: object
+          required:
+            - updated
+            - skipped
+          properties:
+            updated:
+              type: array
+              items:
+                $ref: '../../model/rule_schema/rule_schemas.schema.yaml#/components/schemas/RuleResponse'
+            skipped:
+              type: array
+              items:
+                $ref: '#/components/schemas/SkippedRuleUpgrade'
+        errors:
+          type: array
+          items:
+            $ref: '../../model/error_schema.schema.yaml#/components/schemas/ErrorSchema'
```

Comment:
Shouldn't it be aggregated error?

https://github.com/elastic/kibana/blob/329e696f32eb8426978f96642d24700b0c10cef3/x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/aggregated_prebuilt_rules_error.ts#L8-L15

ErrorSchema is pretty different:

https://github.com/elastic/kibana/blob/329e696f32eb8426978f96642d24700b0c10cef3/x-pack/plugins/security_solution/common/api/detection_engine/model/error_schema.gen.ts#L21-L33